src/v/cloud_topics/level_zero/stm/ctp_stm.cc

```diff
@@ -96,6 +96,14 @@ ss::future<bool> ctp_stm::sync_in_term(ss::abort_source& as) {
     co_return true;
 }
 
+std::optional<cluster_epoch> ctp_stm::estimate_inactive_epoch() const noexcept {
+    auto epoch = _state.estimate_min_epoch();
+    if (!epoch.has_value()) {
+        return std::nullopt;
+    }
+    return prev_cluster_epoch(*epoch);
```

Comment:
nit: if you want to be fancy I think you can do: `return _state.estimate_min_epoch().transform(prev_cluster_epoch);`

Reply:
neat